src/rust/src/x509/verify.rs

```diff
@@ -290,22 +313,30 @@ impl PyClientVerifier {
             py_chain.append(c.extra())?;
         }
 
-        // NOTE: These `unwrap()`s cannot fail, since the underlying policy
-        // enforces the presence of a SAN and the well-formedness of the
-        // extension set.
-        let leaf_san = &chain[0]
-            .certificate()
-            .extensions()
-            .ok()
-            .unwrap()
-            .get_extension(&SUBJECT_ALTERNATIVE_NAME_OID)
-            .unwrap();
-
-        let leaf_gns = leaf_san.value::<SubjectAlternativeName<'_>>()?;
-        let py_gns = parse_general_names(py, &leaf_gns)?;
+        let subjects = {
+            // NOTE: The `unwrap()` cannot fail, since the underlying policy
+            // enforces the well-formedness of the extension set.
+            let leaf_san_ext = &chain[0]
+                .certificate()
+                .extensions()
+                .ok()
+                .unwrap()
+                .get_extension(&SUBJECT_ALTERNATIVE_NAME_OID);
+
+            match leaf_san_ext {
+                None => None,
+                Some(leaf_san) => {
+                    let leaf_gns = leaf_san
+                        .value::<SubjectAlternativeName<'_>>()
+                        .map_err(|e| -> CryptographyError { e.into() })?;
+                    let py_gns = parse_general_names(py, &leaf_gns)?;
+                    Some(py_gns)
+                }
+            }
+        };
```

Comment:
```suggestion
        // NOTE: The `unwrap()` cannot fail, since the underlying policy
        // enforces the well-formedness of the extension set.
        let subjects = match &chain[0]
            .certificate()
            .extensions()
            .ok()
            .unwrap()
            .get_extension(&SUBJECT_ALTERNATIVE_NAME_OID) {
                Some(leaf_san) => {
                    let leaf_gns = leaf_san
                        .value::<SubjectAlternativeName<'_>>()?;
                    Some(parse_general_names(py, &leaf_gns)?)
                }
                None => None,
            };
```